pysilcam/tracking/track.py

```diff
@@ -0,0 +1,375 @@
+                                    maxlength=maxlength)
+    return data, tracks
+
+
+def track_process(configfile, datapath, offset=0):
+    '''
+    This takes a silcam config file (and uses information within the [Tracking] section to track particles in images in
+    datapath. This will produce a -TRACKS.h5 file in the [General]datafile location (in a similar manner to normal
+    silcam processing).
+
+    Functions like pysilcam.tracking.track.make_boxplot can make plots with these -TRACK.h5 files.
+
+    :param configfile:
+    :param datapath:
+    :param offset:
+    :return:
+    '''
+    settings = PySilcamSettings(configfile)
+
+    sctr = sctracker.Tracker()
+    sctr.av_window = settings.Background.num_images
+    sctr.MIN_LENGTH = settings.Tracking.min_length
+    sctr.MIN_SPEED = settings.Tracking.min_speed
+    sctr.cross_correlation_threshold = settings.Tracking.cross_correlation_threshold
+    sctr.THRESHOLD = settings.Process.threshold
+    sctr.ecd_tolerance = settings.Tracking.ecd_tolerance
+    sctr.PIX_SIZE = settings.PostProcess.pix_size
+
+    sctr.path = datapath
+    dataset_name = os.path.split(datapath)[-1] + '-TRACKS'
+    sctr.DATAFILE = os.path.join(settings.General.datafile, dataset_name)
+    sctr.track_length_limit = settings.Tracking.track_length_limit
+    sctr.initialise()
+    sctr.files = sctr.files[offset:]
+
+    os.makedirs(settings.General.datafile, exist_ok=True)
+
+    # setup HDF5 file and metadata
+    print('* Setting up', sctr.DATAFILE + '.h5')
+    with h5py.File(sctr.DATAFILE + '.h5', "a") as HDF5File:
+        meta = HDF5File.require_group('Meta')
+        meta.attrs['Modified'] = str(pd.datetime.now())
+        settings_dict = {s: dict(settings.config.items(s)) for s in settings.config.sections()}
+        meta.attrs['Settings'] = str(settings_dict)
+
+    sctr.process()
+
+def make_boxplot(tracksfile, plotfilename):
+    '''
+    given the input tracksfile (-TRACKs.h5), this function will look for all the other -TRACKS.h5 files in the same
+    folder and use this to create summary boxplots of tracked data, which is saved to the General.datafile location
+    specified within the settings contained in the last -TRACKS.h5 file.
+
+    :param tracksfile:
+    :return:
+    '''
+    h5filedir = os.path.split(tracksfile)[0]
+    h5file_list = glob.glob(os.path.join(h5filedir, '*-TRACKS.h5'))
+    dataset_names = [os.path.split(k)[-1].replace('-TRACKS.h5', '') for k in h5file_list]
+
+    tracks = dict()
+    print('* Loading data:')
+    for f, k in zip(h5file_list, dataset_names):
+        print('  ', f)
+        try:
+            tracks[k] = pd.read_hdf(f, 'Tracking/tracks')
+        except KeyError:
+            settings_tmp = settings_from_h5(f)
+            data, tracks_tmp = load_and_process(f,
+                                            settings_tmp.PostProcess.pix_size,
```

Comment:
E128: continuation line under-indented for visual indent  <!-- Automatically posted by Lintly -->